docs/DeveloperGuide.md

```diff
@@ -382,10 +448,51 @@ testers are expected to do more *exploratory* testing.
 
 1. _{ more test cases …​ }_
 
-### Saving data
+### Getting contacts who are free at specified time and date
+1. Assuming that all contacts do not have a schedule.
+   1. Test case: `freeSchedule ti/10:00 da/2022-05-01`<br>
+      Expected: No contacts displayed.
+   
+   2. Test case: `freeSchedule ti/10:00`
+      Expected: Same result as previous.
+
+2. Assuming that some contacts have a schedule.
+   Situation 1: All contacts are free today at 10am.
+   1. Test case: `freeSchedule ti/10:00`<br>
+      Expected: Displays all contacts with a schedule.
+   
+   Situation 2: All contacts are not free today at specified time.
+   1. Test case: `freeSchedule ti/10:00`<br>
+      Expected: No contacts displayed.
+   
+   Situation 3: Some contacts are free today at specified time.
+   1. Test case: `freeSchedule ti/10:00`<br>
+      Expected: Contacts who are free today at 10am displayed.
+   2. Test case: `freeSchedule ti/08:00 ti/10:00`<br>
+      Expected: Contacts who are free today at 10am will be displayed.
+   3. Test case: `freeSchedule ti/200:00 ti/10:00`<br>
+      Expected: Same result as previous.
+
+   Situation 4. Giving a date that is in the past.
+   1. Test case: `freeSchedule ti/10:00 da/2000-01-01`<br>
+      Expected: No contacts displayed.
+
+3. Assuming that some contacts have recurring events.
+   Situation 1: Contacts have daily events.
+   1. Test case: 
+
+4. Wrong inputs
+   1. Test case: `freeSchedule ti/1000`
+      Expected: Nothing happens. Error message written in result box.
+   2. Test case: `freeSchedule ti/10:00 da/01 Dec 2022`
+      Expected: Same result as previous.
+   
+_{ more test cases to be added }_
+
+#### Saving data
 
 1. Dealing with missing/corrupted data files
 
-   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
+1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
 
-1. _{ more test cases …​ }_
+1. _{ more test cases …​ }_free 
```

Comment:
I think I accidentally typed this in. Can remove when we merge PRs or when we are finalizing it